flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -44,57 +52,66 @@ typedef AdditionalInfoExtractor = Map<String, dynamic>? Function(
 /// )
 /// ```
 ///
-/// See the constructor docs for the argument documentation.
+/// The option [enableAutoTransactions] is enabled by default. For every new
+/// route a transaction is started. It's automatically finished after
+/// [autoFinishAfter] duration or when all child spans are finished,
+/// if those happen to take longer. The transaction will be set to [Scope.span]
+/// if the latter is empty.
+///
+/// Enabling the [setRouteNameAsTransaction] option overrides the current
+/// [Scope.transaction] which will also override the name of the current
+/// [Scope.span]. So be careful when this is used together with performance
+/// monitoring.
 ///
 /// See also:
 ///   - [RouteObserver](https://api.flutter.dev/flutter/widgets/RouteObserver-class.html)
 ///   - [Navigating with arguments](https://flutter.dev/docs/cookbook/navigation/navigate-with-arguments)
 class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
-  /// The option [enableAutoTransactions] is enabled by default.
-  /// For every new route a transaction is started. It's automatically finished
-  /// after [autoFinishAfter] duration or when all child spans are
-  /// finished, if those happen to take longer.
-  /// The transaction will be set to [Scope.span] if the latter is empty.
-  ///
-  /// Enabling the [setRouteNameAsTransaction] option overrides the
-  /// current [Scope.transaction] which will also override the name of the current
-  /// [Scope.span]. So be careful when this is used together with performance
-  /// monitoring.
   SentryNavigatorObserver({
     Hub? hub,
     bool enableAutoTransactions = true,
     Duration autoFinishAfter = const Duration(seconds: 3),
     bool setRouteNameAsTransaction = false,
     RouteNameExtractor? routeNameExtractor,
     AdditionalInfoExtractor? additionalInfoProvider,
+    @internal TimeToDisplayTracker? timeToDisplayTracker,
   })  : _hub = hub ?? HubAdapter(),
-        _enableAutoTransactions = enableAutoTransactions,
-        _autoFinishAfter = autoFinishAfter,
         _setRouteNameAsTransaction = setRouteNameAsTransaction,
         _routeNameExtractor = routeNameExtractor,
-        _additionalInfoProvider = additionalInfoProvider,
-        _native = SentryFlutter.native {
+        _additionalInfoProvider = additionalInfoProvider {
     if (enableAutoTransactions) {
-      // ignore: invalid_use_of_internal_member
       _hub.options.sdk.addIntegration('UINavigationTracing');
     }
+    _timeToDisplayTracker = timeToDisplayTracker ??
```

Comment:
`h`: I don't fully understand how the autoFinishAfter now works if you pass in a `timeToDisplayTracker`. How can we ensure that the `TimeToDisplayTracker` passed here uses the correct parameters for `autoFinishAfter` and `enableAutoTransactions`?